abi.go

```diff
@@ -44,25 +50,42 @@ func newMapABIFromFd(fd *internal.FD) (string, *MapABI, error) {
 
 	return "", &MapABI{
 		MapType(info.mapType),
+		MapID(info.id),
 		info.keySize,
 		info.valueSize,
 		info.maxEntries,
 		info.flags,
+		internal.CString(info.mapName[:]),
+		UnspecifiedProgram,
```

Comment:
I think this is problematic: newMapABIFromFd and newMapABIFromProc should produce the same result for a given fd. Otherwise the behaviour of MapABI will change subtly on different kernels. Maybe we should add that as a test.

Reply:
I don't think there currently is a way to get `OwnerProgType` from `newMapABIFromFd` unfortunately.